### ISSUE TITLE
Workspace extensions can't import workspace packages

### ISSUE THREAD

**Issue by emahuni** [directus/directus#20152]
### Describe the Bug

I have this form-scripts extension that I have developed. There is a discussion about it that I opened up some months ago https://github.com/directus/directus/discussions/19053 that details what it's all about, and have a working implementation on a modified @directus/app and now this new extension; I managed to simplify this into an extension so that it can be installed on any Directus installation without modifying `@directus/app`. The extension is a bundle with an interface and endpoint, and it works as intended when `SERVE_APP` is `false`, and used with a customized  `@directus/app`. 

the following is the workspace setup that causes this issue:
![image](https://github.com/directus/directus/assets/7290508/9c99b6b8-51ac-4b58-b37e-6470efae8898)

`extensions` are not local extensions, they are fully independent extension packages that I intend on sharing with the community. The main reason was to demonstrate to Directus team, what I so far seen as important parts of this feature.

So using workspaces I can refer to them this way in the api package (self-hosted setup):

![image](https://github.com/directus/directus/assets/7290508/e488df0a-3704-494a-b8fa-bcd9aa340432)

just like a normal package.

There is another folder in the api folder (where the self-hosted Directus app is hosted) that serves the local extensions purposes, which has nothing to do with this.

The packages that are imported by the extension are in `packages` eg: `packages/@emahuni/stdlib`, that is simply imported in the extension as `import stdlib from '@emahuni/stdlib'` as it is the full name of the package.

I am running into issues when the extension imports other workspace packages. All extensions fail to load because of that import issue in the extension (though this works well if I DON'T SERVE_APP from the API and run a separate app serve). 

![image](https://github.com/directus/directus/assets/7290508/250f9b64-9f2e-4956-9b4b-94fc2d1ad40c)
Which in sources is: 
![image](https://github.com/directus/directus/assets/7290508/1ed78cd7-d4e5-4333-8f87-28cd2602b339)

 It's related to rollup's chunking, found some code in the `extensions.ts` file in the api that is referred from the app:
![image](https://github.com/directus/directus/assets/7290508/906cd5b8-bd84-4c10-81ef-425a8be3018f)

### To Reproduce

1. Create a self-hosted setup with the above mentioned workspaces structure
- just 3 dirs will be enough; api, extensions and packages
- directus will be self-hosted in `api` package as an npm package using `pnpm add directus@10.6.3`
- test extension will be created in `extensions ` dir
- test package will be create in `packages `dir.

3. Add some extensions from npm registry
4. Start the server in api => `pnpm directus start`
5. Create a collection that uses the extensions
6. Create a new item for the collection
7. They should work as intended
8. Stop the server
9. Create an interface extension package in the workspaces extensions dir eg; `extensions/test_ext`, named: `directus-extension-test-ext`
10. Create a package in the packages eg: `test-pkg`
11. Add the extension to the self-hosted api using `pnpm add directus-extension-test-ext`
12. Add package to the extension using `pnpm add test-pkg` and import it for use in the extension's `interface.vue` file.
13. build the extension
14. `pnpm directus start`
15. The extension should be loaded with others successfully
16. Use the new workspace interface extension in the earlier collection together with the other extensions
17. Create a new item for the collection
18. The extensions should fail to load and the above-mentioned error should appear in the browser console.

### Directus Version

v10.6.3 (tried this on the latest 10.7.0, same story)

### Hosting Strategy

Self-Hosted (Custom)

**Comment by br41nslug** [comment]
Tbh this sounds more like a monorepo issue 😬 Is the package in the workspace getting bundled properly into the resulting extension?

**Comment by br41nslug** [comment]
I have created a most basic pnpm mono repo with:
- a directus instance
- a custom interface (used in directus)
- a npm package (used in the interface)

I am not able to reproduce your error with the provided information. It does not seem to be directly related to having `workspace:*` dependencies. The issue is likely in something specific done by the extension or package you've mentioned.

Closing this for now, but happy to re-open it when more information becomes available.
